src/main_thread/init/media_source_content_initializer.ts

```diff
@@ -708,6 +712,34 @@ export default class MediaSourceContentInitializer extends ContentInitializer {
             ? !playbackObserver.getIsPaused()
             : autoPlay;
           onReloadOrder({ position, autoPlay: autoplay });
+        };
```

Comment:
I think I've done that because as a closure, it would then have an implicit dependency on the time at which it should be called: i.e. it should be called ASAP - and probably synchronously, as at the time of the next observation it would rely on stale data.

To avoid adding such complex implicit dependencies, I get the last observation, even if right now it's only called synchronously so that last observation is the same one than in the parent scope.

Basically to me: there's nothing in that function declaration indicating that we should call it synchronously. Thus by relying on the parent scope's observation, we risk creating future bugs (or more probably: future weird behaviors where we would be reloading in the "past").